ehr/resources/web/ehr/panel/TaskDataEntryPanel.js

```diff
@@ -20,7 +20,11 @@ Ext4.define('EHR.panel.TaskDataEntryPanel', {
         cfg = this.callParent(arguments);
         cfg.filterArray = cfg.filterArray || [];
         cfg.filterArray.push(LABKEY.Filter.create('taskid', this.taskId, LABKEY.Filter.Types.EQUALS));
+        // A redundant filter that can help the DB choose a more efficient query plan. SQLServer does better with this
+        // filter in choosing to use the index on the TaskId column, even if the value is being passed as a JDBC
+        // parameter instead of being embedded in the SQL as a string literal.
+        cfg.filterArray.push(LABKEY.Filter.create('taskid', this.requestId, LABKEY.Filter.Types.NOT_MISSING));
```

Comment:
The second param shouldn't matter for the NOT_MISSING filter type, but this one should like either be `this.taskId`, for consistency, or `null`.

Reply:
Indeed. Copy/paste error now corrected